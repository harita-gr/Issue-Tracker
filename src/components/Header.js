import React from 'react'
import {LinkContainer} from 'react-router-bootstrap'
import {Container,Navbar,Nav,Button} from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux'
import {logout} from '../redux/actions/userActions'
function Header() {

    const dispatch = useDispatch()
    const userLogin = useSelector((state) => state.userLogin)
    var { userInfo } = userLogin

    const logoutHandler = () => {
        dispatch(logout())
      }

    return (
        <header>          
            <Navbar bg="danger" expand="lg">
             <Container>
                <LinkContainer to='/'>
                    <Navbar.Brand className="brand">Issue Tracker</Navbar.Brand>
                </LinkContainer>

                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav justify-content-end">
                    <Nav className="ml-auto">
                    <LinkContainer to='/about'>
                    <Nav.Link className="nav-link"> ABOUT</Nav.Link>
                    </LinkContainer>
                    <LinkContainer to='/issues'>
                    <Nav.Link className="nav-link"> ISSUES</Nav.Link>
                    </LinkContainer>
                    <LinkContainer to='/reports'>
                    <Nav.Link className="nav-link"> REPORTS</Nav.Link>
                    </LinkContainer>
                    {
                        userInfo && 
                        <LinkContainer to='/signin'>
                        <Button variant="dark" className="login-btn" onClick={logoutHandler} ><i className="fas fa-user"></i> LOGOUT</Button>  
                      </LinkContainer>
                    }
                    {
                        !userInfo && 
                    <LinkContainer to='/signin'>
                      <Button variant="dark" className="login-btn" ><i className="fas fa-user"></i> SIGN IN</Button>  
                    </LinkContainer>
                    }
                    
                    </Nav>

                </Navbar.Collapse>
            </Container>   
            </Navbar>
        </header>
    )
}

export default Header
