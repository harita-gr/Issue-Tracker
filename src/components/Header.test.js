import Header from './Header'
import {BrowserRouter as Router} from 'react-router-dom'
import React from 'react';
import {shallow } from 'enzyme';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render } from '@testing-library/react'


describe('<Header/>', () => {

    let wrapper;
    const initialState = {userLogin: { userInfo: 'dummyData'}}
    const mockStore = configureStore()
    let store;
  

    beforeEach( () =>{
        store = mockStore(initialState)
        wrapper = render(<Provider store={store}><Router><Header/></Router></Provider>); 
        //console.log(wrapper.debug())
    })

        it('render the 3 nav links',() => {
            const { getByText } = wrapper
            expect(getByText('ISSUES')).not.toBeNull()
            expect(getByText('REPORTS')).not.toBeNull()
            expect(getByText('ABOUT')).not.toBeNull()
            
        })

        it('render the brand title in nav bar',() => {
          const { getByText } = wrapper
          expect(getByText('Issue Tracker')).not.toBeNull()
          
      })

      it('render the header correctly',() =>{      
        const container = render(<Provider store={store}><Router><Header/></Router></Provider>)
        expect(container).toMatchSnapshot();
      })

      afterEach( () =>{
        jest.clearAllMocks();
      })


});