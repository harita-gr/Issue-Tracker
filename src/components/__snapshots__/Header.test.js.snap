// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> render the header correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav
          class="navbar navbar-expand-lg navbar-light bg-danger"
        >
          <div
            class="container"
          >
            <a
              class="brand active navbar-brand"
              href="/"
            >
              Issue Tracker
            </a>
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="navbar-collapse collapse"
              id="basic-navbar-nav justify-content-end"
            >
              <div
                class="ml-auto navbar-nav"
              >
                <a
                  class="nav-link nav-link"
                  data-rb-event-key="/about"
                  href="/about"
                >
                   ABOUT
                </a>
                <a
                  class="nav-link nav-link"
                  data-rb-event-key="/issues"
                  href="/issues"
                >
                   ISSUES
                </a>
                <a
                  class="nav-link nav-link"
                  data-rb-event-key="/reports"
                  href="/reports"
                >
                   REPORTS
                </a>
                <a
                  class="login-btn btn btn-dark"
                  href="/signin"
                >
                  <i
                    class="fas fa-user"
                  />
                   LOGOUT
                </a>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <div>
      <header>
        <nav
          class="navbar navbar-expand-lg navbar-light bg-danger"
        >
          <div
            class="container"
          >
            <a
              class="brand active navbar-brand"
              href="/"
            >
              Issue Tracker
            </a>
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="navbar-collapse collapse"
              id="basic-navbar-nav justify-content-end"
            >
              <div
                class="ml-auto navbar-nav"
              >
                <a
                  class="nav-link nav-link"
                  data-rb-event-key="/about"
                  href="/about"
                >
                   ABOUT
                </a>
                <a
                  class="nav-link nav-link"
                  data-rb-event-key="/issues"
                  href="/issues"
                >
                   ISSUES
                </a>
                <a
                  class="nav-link nav-link"
                  data-rb-event-key="/reports"
                  href="/reports"
                >
                   REPORTS
                </a>
                <a
                  class="login-btn btn btn-dark"
                  href="/signin"
                >
                  <i
                    class="fas fa-user"
                  />
                   LOGOUT
                </a>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-danger"
      >
        <div
          class="container"
        >
          <a
            class="brand active navbar-brand"
            href="/"
          >
            Issue Tracker
          </a>
          <button
            aria-controls="basic-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="basic-navbar-nav justify-content-end"
          >
            <div
              class="ml-auto navbar-nav"
            >
              <a
                class="nav-link nav-link"
                data-rb-event-key="/about"
                href="/about"
              >
                 ABOUT
              </a>
              <a
                class="nav-link nav-link"
                data-rb-event-key="/issues"
                href="/issues"
              >
                 ISSUES
              </a>
              <a
                class="nav-link nav-link"
                data-rb-event-key="/reports"
                href="/reports"
              >
                 REPORTS
              </a>
              <a
                class="login-btn btn btn-dark"
                href="/signin"
              >
                <i
                  class="fas fa-user"
                />
                 LOGOUT
              </a>
            </div>
          </div>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
